### PR TITLE
Finetuning: load slim checkpoints

### PR DESCRIPTION
**Pull request by nostalgebraist**
Allows you to pass a checkpoint without an optimizer as the `--tune-model-path` for `device_train`.

(Obviously, this only makes sense if you want to use a fresh optimizer with `--fresh-opt`.)

----

Speeds up load times by about 5x:

```
read from disk/gcs in 116.074s
network loaded in 124.724s
```

vs

```
read from disk/gcs in 602.934s
network loaded in 650.299s
```

Does not seem to adversely impact the loss/quality of the fine-tuned model, as far as I can tell.

### CODE REVIEW

**Comment by kingoflolz** [comment]
Thanks!
